core/src/main/java/io/grpc/ServerInterceptors.java

```diff
@@ -113,4 +161,89 @@ private InterceptCallHandler(ServerInterceptor interceptor,
       return interceptor.interceptCall(method, call, headers, callHandler);
     }
   }
+
+  private static <OReqT, ORespT, WReqT, WRespT> ServerMethodDefinition<WReqT, WRespT> wrapMethod(
+      final ServerMethodDefinition<OReqT, ORespT> definition,
+      final MethodDescriptor<WReqT, WRespT> wrappedMethod) {
+    final ServerCallHandler<WReqT, WRespT> wrappedHandler = wrapHandler(
+        definition.getServerCallHandler(),
+        definition.getMethodDescriptor(),
+        wrappedMethod);
+    return ServerMethodDefinition.create(wrappedMethod, wrappedHandler);
+  }
+
+  private static <OReqT, ORespT, WReqT, WRespT> ServerCallHandler<WReqT, WRespT> wrapHandler(
+      final ServerCallHandler<OReqT, ORespT> originalHandler,
+      final MethodDescriptor<OReqT, ORespT> originalMethod,
+      final MethodDescriptor<WReqT, WRespT> wrappedMethod) {
+    return new ServerCallHandler<WReqT, WRespT>() {
+      @Override
+      public ServerCall.Listener<WReqT> startCall(
+          final MethodDescriptor<WReqT, WRespT> method,
+          final ServerCall<WRespT> call,
+          final Metadata headers) {
+        final ServerCall<ORespT> unwrappedCall = new ServerCall<ORespT>() {
```

Comment:
Yea, I tried that at first then realized I couldn't properly convert things :(  I was thinking there might be a way to get tricky using Object and lots of casting to pull it off, but nothing seemed obvious in the end.  One option is create a ForwardingAndConvertingServerCall potentially which would exist aside the ForwardingServerCall, and we could potentially make this one step more generic by having the "useByteArrayMessages" instead become a <T> convertMessages(MethodDescriptor<T>) type method instead?